File: 0025 - Reverse Nodes in k-Group/solution.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* reverseKGroup(struct ListNode* head, int k){
    struct ListNode *now = head, *prev, *tmp;
    for(int i = 0; i < k; i++){
        if(!now) return head;
        now = now->next;
    }
    prev = reverseKGroup(now, k), now = head;
    for(int i = 0; i < k; i++){
        tmp = now->next;
        now->next = prev;
        prev = now;
        now = tmp;
    }
    return prev;
}

File: 0025 - Reverse Nodes in k-Group/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseKGroup(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        now, now.next, n = self, head, 0
        while now.next:
            now = now.next
            n += 1
        now = self
        for i in range(n//k):
            head, tail, now = now, now.next, now.next.next
            for j in range(k-1):
                tail.next = now.next
                now.next = head.next
                head.next = now
                now = tail.next
            now = tail
        return self.next